pkg/issuer/vault/setup.go

```diff
@@ -60,7 +60,9 @@ func (v *Vault) Setup(ctx context.Context) error {
 	// check if at least one auth method is specified.
 	if v.issuer.GetSpec().Vault.Auth.TokenSecretRef.Name == "" &&
 		v.issuer.GetSpec().Vault.Auth.AppRole.RoleId == "" &&
-		v.issuer.GetSpec().Vault.Auth.AppRole.SecretRef.Name == "" {
+		v.issuer.GetSpec().Vault.Auth.AppRole.SecretRef.Name == "" &&
+		v.issuer.GetSpec().Vault.Auth.Kubernetes.Role == "" &&
+		v.issuer.GetSpec().Vault.Auth.Kubernetes.SecretRef.Name == "" {
```

Comment:
Instead of checking particular fields of these different auth types, we should make each auth type structure a pointer and then just check if they are nil here. This is super messy right now and could really confuse users IMO.